test/e2e/unidling.go

```diff
@@ -199,6 +199,23 @@ var _ = ginkgo.Describe("Unidling", func() {
 				return hittingGeneratesNewEvents(service, cs, clientPod, cmd)
 			}, 10*time.Second, 1*time.Second).Should(gomega.Equal(true), "New events are not generated")
 		})
+
+		ginkgo.It("Should connect to an unidled backend at the first attempt", func() {
+
+			// Simulate service unidling
+			_, err := jig.UpdateService(func(service *v1.Service) {
+				service.Annotations = nil
+			})
+			framework.ExpectNoError(err)
+			go func() {
+				defer ginkgo.GinkgoRecover()
```

Comment:
@zeeke goroutines can sometimes take a while to start up, so this could be prone to races and cause flakes.

I would use a waitgroup here to synchronize:

```
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer ginkgo.GinkgoRecover()
		wg.Done()
		time.Sleep(time.Second)
		createBackend(f, serviceName, namespace, node, jig.Labels, port)
	}()

	// Connecting to the service should work at the first attempt
	wg.Wait()
	gomega.Expect(checkService(clientPod, cmd)).To(gomega.Equal(works))
```

This will make sure that the time.Sleep() and the Expect() start at the same time.

Reply:
sure! Thanks for catching that